### PR TITLE
DAML-LF: Kill RelativeContractId

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_END
CHANGELOG_BEGIN

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueSpec.scala [hunk 1]
```diff
@@ -140,8 +116,8 @@ class ValueSpec
   }
 
   "AbsoluteContractId" - {
```
Comment (SamirTalwar-DA) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/value/ValueSpec.scala:118]: You might want to change this test name (and others) in this file.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala [hunk 2]
```diff
@@ -105,10 +105,10 @@ package object v1 {
     * absolute contract identifiers.
     */
   type CommittedTransaction =
-    GenTransaction.WithTxValue[NodeId, Value.AbsoluteContractId]
+    GenTransaction.WithTxValue[NodeId, Value.ContractId]
 
   /** A contract instance with absolute contract identifiers only. */
   type AbsoluteContractInst =
```
Comment (SamirTalwar-DA) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala:111]: Do you plan on renaming this one too?

**Comment by gerolf-da** [comment]
@remyhaemmerle-da: Removing `RelativeContractId` should also mean we can remove `BlindingInfo#localDivulgence` and everything related to it.

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala [hunk 3]
```diff
@@ -344,30 +339,19 @@ object Value extends CidContainer1WithDefaultCidResolver[Value] {
...
-    * Note that relative contract ids are useful only before commit, in
-    * the context of a transaction. After committing we should never
-    * mention them.
-    *
-    * Why put it here and not just in Transaction.scala? Because we want
-    * to be able to use AbsoluteContractId elsewhere, so that we can
-    * automatically upcast to ContractId by subtyping.
-    */
-  sealed abstract class ContractId extends Product with Serializable
+  @deprecated("Use ContractId", since = "1.1.2")
```
Comment (miklos-da) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala:342]: What's our process for removing deprecated elements such as this? Do we remove them in e.g. the next major SDK version?

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala [hunk 4]
```diff
@@ -105,10 +105,10 @@ package object v1 {
     * absolute contract identifiers.
```
Comment (miklos-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala:105]: Please refine the comments above as well (they still reference relative and absolute contract identifiers).

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala:342]: We do not have clearly defined process. My intention was to let this deprecated warning for a least one release.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala]: done.

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala (hunk 3 above)
Comment (miklos-da) [reply daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala:342]: Sounds good.

File: daml-lf/spec/contract-id.rst [hunk 5]
```diff
@@ -206,13 +206,13 @@ The submission performs the following steps:
   discriminator is not `fresh <Discriminator Freshness_>`_, abort the
   interpretation. The submitter can restart the interpretation, which will pick
   another submission seed.
-* If the transaction succeeds, the output is a *raw transaction*
+* If the transaction succeeds, the output is a *submitted transaction*
 
 
 Depending of the ledger implementation, the local contract IDs are
```
Comment (miklos-da) [inline daml-lf/spec/contract-id.rst:212]:
```suggestion
Depending on the ledger implementation, the local contract IDs are
```

File: daml-lf/spec/contract-id.rst [hunk 6]
```diff
@@ -206,13 +206,13 @@ The submission performs the following steps:
...
-suffixed with a suffix in a latter step. This yields the *ready
-transaction*. For ledgers that do not require suffixing, raw and ready
-transaction coincide. Ready transactions are the source of true to
+suffixed with a suffix in a latter step. This yields the *committed
```
Comment (miklos-da) [inline daml-lf/spec/contract-id.rst:213]:
```suggestion
suffixed with a suffix in a later step. This yields the *committed
```

File: daml-lf/spec/contract-id.rst [hunk 7]
```diff
@@ -206,13 +206,13 @@ The submission performs the following steps:
...
+transaction*. For ledgers that do not require suffixing, committed and submitted
+transaction coincide. Committed transactions are the source of true to
```
Comment (miklos-da) [inline daml-lf/spec/contract-id.rst:215]:
```suggestion
transactions coincide. Committed transactions are the source of truth to
```

File: daml-lf/spec/contract-id.rst [hunk 8]
```diff
@@ -206,13 +206,13 @@ The submission performs the following steps:
...
 describe the state if the ledger.
```
Comment (miklos-da) [inline daml-lf/spec/contract-id.rst:216]:
```suggestion
derive the state of the ledger.
```

**Pull request merged**